app/src/androidTest/java/com/epfl/drawyourpath/database/MockNonWorkingDatabaseTest.kt

```diff
@@ -0,0 +1,36 @@
+package com.epfl.drawyourpath.database
+
+import com.epfl.drawyourpath.userProfile.dailygoal.DailyGoal
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+
+class MockNonWorkingDatabaseTest {
+
+    @Test
+    fun everyFunctionShouldThrowError() {
+        val mock = MockNonWorkingDatabase()
+        val mockUser = MockDatabase().mockUser
+
+        mock.isUserInDatabase("").assertError(true)
+        mock.getUsername("").assertError("")
+        mock.getUserIdFromUsername("").assertError("")
+        mock.isUsernameAvailable("").assertError(true)
+        mock.setUsername("", "").assertError(Unit)
+        mock.setUserData("", UserData()).assertError(Unit)
+        mock.getUserData("").assertError(mockUser)
+        mock.setGoals("", UserGoals()).assertError(Unit)
+        mock.addFriend("", "").assertError(Unit)
+        mock.removeFriend("", "").assertError(Unit)
+        mock.addDailyGoal("", DailyGoal(0.0, 0.0, 0)).assertError(Unit)
+        mock.updateUserAchievements("", 0.0, 0.0).assertError(Unit)
+    }
```

Comment:
There is no test for the functions : 
- createUser
- setProfilePhoto
- addRunHistory 
- removeRunHistory

Reply:
It seems there are tests for `addRunHistory` and `removeRunHistory`.

Reply:
There are not tested in the case of the NonWorkingMockDatabase. 
mock.createUser(arguments....).assertError(Unit)
like for the other methods like removeFriend, addDailyGoal....